tests/sqlparser_mysql.rs

```diff
@@ -176,6 +176,33 @@ fn parse_quote_identifiers() {
     }
 }
 
+#[test]
+fn parse_escaped_string() {
+    let sql = r#"SELECT 'I\'m fine'"#;
+
+    let projection = mysql().verified_only_select(sql).projection;
+    let item = projection.get(0).unwrap();
+
+    match &item {
+        SelectItem::UnnamedExpr(Expr::Value(value)) => {
+            assert_eq!("'I\\'m fine'", value.to_string());
+        }
+        _ => unreachable!(),
+    }
+
+    let sql = r#"SELECT 'I''m fine'"#;
+
+    let projection = mysql().verified_only_select(sql).projection;
+    let item = projection.get(0).unwrap();
+
+    match &item {
```

Comment:
Can you please also add a test for a string ending in `\`(unterminated sequence) as well as for a non MySql dialect ?